### ISSUE TITLE
Files loaded though custom VirtualPathProvider not found

### ISSUE THREAD

**Issue by mcintyre321** [andrewdavey/cassette#59]
Hi!

I use a custom VirtualPathProvider (http://nuget.org/List/Packages/EmbeddedResourceVirtualPathProvider) to load resources from assemblies. They don't seem to be picked up.

I suspect that Cassette looks to see if files exist inside the web folder using the filesystem rather than
```System.Web.Hosting.HostingEnvironment.VirtualPathProvider.FileExists(string virtualPath);

``````
and content should be loaded via:
```var stream = HostingEnvironment.VirtualPathProvider.OpenFile(file);
    var text = new StreamReader(stream).ReadToEnd();
``````

I had a very quick look at the source but am not sure how to patch it exactly. Apologies if I'm wrong about this and it's something else I'm doing that is preventing my code from working!

**Comment by andrewdavey** [comment]
Yes, Cassette currently digs into the file system directly. Making it go via the virtual path provider should be feasible, given Cassette has an abstraction (`Cassette.IO.IDirectory` and `IFile`).

I'd certainly welcome a pull request with virtual path provider implementation of those interfaces :) Otherwise I'll try to get around to it at some point. Give me a shout on twitter/email if you'd like more pointers on where to get started.
